workloads/files/workload-network-script-cm.yml

```diff
@@ -0,0 +1,961 @@
+          fi
+          retries=$[$retries + 1]
+        done
+        [ "${client_ready}" == "false" ] && echo "$(date -u) uperf-client-${rc_num} failed to be ready on time" && exit 1
+        retries=0
+        while [ $retries -le 60 ] ; do
+          server_ready=`oc get replicationcontroller/uperf-server-${rc_num} -o json -n network0 | jq '.status.replicas==.status.availableReplicas'`
+          if [ "${server_ready}" == "true" ]; then
+            echo "$(date -u) uperf-server-${rc_num} is ready"
+            break
+          else
+            echo "$(date -u) Waiting for uperf-server-${rc_num} to be ready"
+            sleep 1
+          fi
+          retries=$[$retries + 1]
+        done
+        [ "${server_ready}" == "false" ] && echo "$(date -u) uperf-server-${rc_num} failed to be ready on time" && exit 1
+      done
+
+      if [ "${NETWORK_TEST_SERVICE}" == "true" ]; then
+        clients=`oc get svc --output=json -n network0 --selector="name=client" | jq -r '[.items[].spec.clusterIP] | join(",")'`
+        servers=`oc get svc --output=json -n network0 --selector="name=server" | jq -r '[.items[].spec.clusterIP] | join(",")'`
+      else
+        clients=`oc get pods --output=json -n network0 --selector="name=client" | jq -r '[.items[].status.podIP] | join(",")'`
+        servers=`oc get pods --output=json -n network0 --selector="name=server" | jq -r '[.items[].status.podIP] | join(",")'`
+      fi
+      echo "${clients} ${servers}"
+      mv /root/.ssh/config /root/.ssh/config.bak
+      ssh_config_template="$(cat /root/workload/ssh_config.template)"
+      for host in $(echo "${clients} ${servers}" | sed "s/,/ /g")
+      do
+        eval "echo \"${ssh_config_template}\"" >> /root/.ssh/config
+      done
+      cat /root/.ssh/config.bak >> /root/.ssh/config
+      chmod 0600 /root/.ssh/config
+
+      pbench-uperf --test-types=${NETWORK_TEST_TYPE} \
+        --runtime=${NETWORK_TEST_RUNTIME} \
+        --message-sizes=${NETWORK_TEST_MESSAGE_SIZES} \
+        --protocols=${NETWORK_TEST_PROTOCOL} \
+        --instances=${NETWORK_TEST_INSTANCES} \
+        --samples=${NETWORK_TEST_SAMPLES} \
+        --max-stddev=10 \
+        --clients=${clients} \
+        --servers=${servers} \
+        --config=${config_prefix}_${pair_count}p_${NETWORK_TEST_INSTANCES//,/_}i_${NETWORK_TEST_PROTOCOL/,/_}
+
+      # Clean up project
+      oc delete project network0
+      retries=0
+      while [ $retries -le 60 ] ; do
+        remaining_projects=`oc get projects | egrep "^network[0-9]+ " | wc -l`
+        echo "$(date -u) Deleting network0 project..."
+        if [ $remaining_projects -eq 0 ]; then
+          break
+        fi
+        sleep 1
+        retries=$[$retries + 1]
+      done
+      [ ${remaining_projects} -gt 0 ] && echo "$(date -u) Failed to clean up network0 project" && exit 1
+
+      pbench-copy-results --prefix ${NETWORK_TEST_PREFIX}
+      echo "$(date -u) Completed Network benchmarks with ${pair_count} pairs"
```

Comment:
There seems to be too many of the `echo "$(date -u)"` commands, use a "log()" function to stderr for more consistency?